src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -166,37 +178,72 @@ export class EditorComponent implements OnInit {
       this.scaleFactor =  1;
     }
 
-    //  Canvas to draw mask, hidden.
+    // Canvas to draw mask, hidden.
     this.maskCanvas.nativeElement.width = imgWidth;
     this.maskCanvas.nativeElement.height = imgHeight;
     this.maskCtx = this.maskCanvas.nativeElement.getContext('2d');
     this.maskCtx.lineCap = this.maskCtx.lineJoin = 'round';
     this.maskCtx.strokeStyle = this.MAGENTA;
 
-    //  Canvas to show mask scaled.
+    // Canvas to show mask scaled.
     this.scaledCanvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.scaledCanvas.nativeElement.height = imgHeight * this.scaleFactor;
     this.scaledCtx = this.scaledCanvas.nativeElement.getContext('2d');
 
-    //  Canvas to show Image (never changes unless user only wants to see Mask)
+    // Canvas to show Image (never changes unless user only wants to see Mask)
     this.imageCanvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.imageCanvas.nativeElement.height = imgHeight * this.scaleFactor;
     this.imageCtx = this.imageCanvas.nativeElement.getContext('2d');
-    
+
     this.stageWidth = imgWidth * this.scaleFactor;
     this.stageHeight = imgHeight * this.scaleFactor;
 
-    //   Draws image non scaled on full canvas
+    // Canvas to paint cursor-overlay of brush size.
+    this.cursorCanvas.nativeElement.width = imgWidth * this.scaleFactor;
+    this.cursorCanvas.nativeElement.height = imgHeight * this.scaleFactor;
+
+    // Updates the drawn 'cursor' when the user's mouse moves.
+    this.maskTarget = document.querySelector('#mask-layer');
+    this.cursorTarget = document.querySelector('#cursor-layer');
+    this.cursorCtx = this.cursorTarget.getContext('2d');
+    this.maskTarget.addEventListener('mousemove', (e) => {
+      this.setCursorPosition(e);
+    }, false);
+    this.maskTarget.addEventListener('mouseout', () => {
```

Comment:
I think the event listeners can just go in the mask.directive under the respective event listeners right? You could Input the cursorCtx and setCursorPosition can be called in mask.directive. I think you can pass ctx's through inputs. This way the user interaction is separate from the component.

Reply:
Or you could Output the event and catch it with the setCursorPosition, we can talk about this if you want to!

Reply:
Moved implementation into mask directive